test/extensions/network/dns_resolver/cares/dns_impl_test.cc

```diff
@@ -434,20 +545,50 @@ class DnsImplTest : public testing::TestWithParam<Address::IpVersion> {
   DnsImplTest()
       : api_(Api::createApiForTest()), dispatcher_(api_->allocateDispatcher("test_thread")) {}
 
+  envoy::config::core::v3::TypedExtensionConfig getTypedDnsResolverConfig(
+      const std::vector<Network::Address::InstanceConstSharedPtr>& resolver_inst,
+      const envoy::config::core::v3::DnsResolverOptions& dns_resolver_options) {
+    envoy::config::core::v3::TypedExtensionConfig typed_dns_resolver_config;
+    envoy::extensions::network::dns_resolver::cares::v3::CaresDnsResolverConfig cares;
+    auto dns_resolvers = envoy::config::core::v3::Address();
+
+    // setup and address
+    for (const auto& resolver : resolver_inst) {
+      if (resolver->ip() != nullptr) {
+        dns_resolvers.Clear();
+        dns_resolvers.mutable_socket_address()->set_address(resolver->ip()->addressAsString());
+        dns_resolvers.mutable_socket_address()->set_port_value(resolver->ip()->port());
+        cares.add_resolvers()->MergeFrom(dns_resolvers);
+      }
+    }
+    // copy over dns_resolver_options_
+    cares.mutable_dns_resolver_options()->MergeFrom(dns_resolver_options);
+    // setup the typed config
+    typed_dns_resolver_config.mutable_typed_config()->PackFrom(cares);
+    typed_dns_resolver_config.set_name(std::string(Network::CaresDnsResolver));
+
+    return typed_dns_resolver_config;
+  }
+
   void SetUp() override {
     // Instantiate TestDnsServer and listen on a random port on the loopback address.
     server_ = std::make_unique<TestDnsServer>(*dispatcher_);
     socket_ = std::make_shared<Network::Test::TcpListenSocketImmediateListen>(
         Network::Test::getCanonicalLoopbackAddress(GetParam()));
     listener_ = dispatcher_->createListener(socket_, *server_, true);
     updateDnsResolverOptions();
+
     // Create a resolver options on stack here to emulate what actually happens in envoy bootstrap.
     envoy::config::core::v3::DnsResolverOptions dns_resolver_options = dns_resolver_options_;
+    auto typed_dns_resolver_config_in_construct = getTypedDnsResolverConfig(
+        {socket_->connectionInfoProvider().localAddress()}, dns_resolver_options);
+    auto typed_dns_resolver_config_not_in_construct =
+        getTypedDnsResolverConfig({}, dns_resolver_options);
```

Comment:
I think these can be moved to their corresponding if-else scopes.

Reply:
ACK